### DISCUSSION TITLE
Implementing skip connections in GCNs

### DISCUSSION THREAD

**Discussion by codeastra2** [pyg-team/pytorch_geometric#6902] [Ideas]
Skip connections are a useful feature to overcome issues of over-squashing: 
1. https://minyoungg.github.io/MIT-deeplearning-blogs/2021/12/09/oversquashing-in-gnns/
2. https://arxiv.org/pdf/1806.03536.pdf 

When aggregating some of the information from the nodes we skip some of these nodes, could we add a argument in GCN(in_f, out_f, skip=True) which would do this?

**Comment by rusty1s** [comment]
I think we already provide this option via the `jk` flag, *e.g.*, in `GCN` or `SAGEConv`. There also exists a separate `JumpingKnowledge` module.

**Comment by codeastra2** [comment]
@rusty1s  thanks for your reply, I searched the documentation of 
1. https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#convolutional-layers 
2. https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/message_passing.html
and even GraphSage but did not find "jk" operator, could you please let me know the same.

**Reply by rusty1s** [reply]
See https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.models.GraphSAGE.html#torch_geometric.nn.models.GraphSAGE
